kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/ProxyChannelState.java

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal;
+
+import java.util.List;
+import java.util.Objects;
+
+import org.apache.kafka.common.message.ApiVersionsRequestData;
+
+import io.netty.handler.codec.haproxy.HAProxyMessage;
+
+import io.kroxylicious.proxy.filter.NetFilter;
+import io.kroxylicious.proxy.frame.DecodedRequestFrame;
+import io.kroxylicious.proxy.service.HostPort;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+import static io.kroxylicious.proxy.internal.ProxyChannelState.ApiVersions;
+import static io.kroxylicious.proxy.internal.ProxyChannelState.ClientActive;
+import static io.kroxylicious.proxy.internal.ProxyChannelState.Closed;
+import static io.kroxylicious.proxy.internal.ProxyChannelState.Connecting;
+import static io.kroxylicious.proxy.internal.ProxyChannelState.Forwarding;
+import static io.kroxylicious.proxy.internal.ProxyChannelState.HaProxy;
+import static io.kroxylicious.proxy.internal.ProxyChannelState.SelectingServer;
+import static io.kroxylicious.proxy.internal.ProxyChannelState.Startup;
+
+/**
+ * Root of a sealed class hierarchy representing the states of the {@link ProxyChannelStateMachine}.
+ */
+sealed interface ProxyChannelState permits
+        Startup,
+        ClientActive,
+        HaProxy,
+        ApiVersions,
+        SelectingServer,
+        Connecting,
+        Forwarding,
+        Closed {
+
+    /**
+     * The statemachine has just been created.
+     */
+    record Startup() implements ProxyChannelState {
+        public static final Startup STARTING_STATE = new Startup();
+
+        public ClientActive toCLientActive() {
```

Comment:
method has a sneaky capital L

Reply:
fixed in b2f717fe937bb9ff98a8c992a0abd66dd73ead1f